### PR TITLE
Detect input from variable def statements in transform stmt

### PR DESCRIPTION
**Pull request by maheeka**

### CODE REVIEW

File: modules/ballerina-core/src/main/java/org/ballerinalang/model/builder/BLangModelBuilder.java [hunk 1]
```diff
@@ -2166,7 +2166,24 @@ private void validateIdentifier(String identifier, NodeLocation location) {
...
         for (Statement statement : blockStmt.getStatements()) {
-            if (statement instanceof AssignStmt) {
+            if (statement instanceof VariableDefStmt) {
+                VariableDefStmt variableDefStmt = (VariableDefStmt) statement;
+                Expression varRefExpression = variableDefStmt.getLExpr();
+                String varName = ((SimpleVarRefExpr) varRefExpression).getVarName();
+                //variables defined in transform scope, cannot be used as output
+                if (outputs.get(varName) == null) {
+                    //if variable has not been used as an output before
+                    if (inputs.get(varName) == null) {
```
Comment (lankavitharana) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/builder/BLangModelBuilder.java]: Shall we refactor this to something like 
if(abc == null) {
   return;
}
not null logic...
if(cde == null) {
   return;
}
not null logic

Comment (maheeka) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/builder/BLangModelBuilder.java]: Fixed with https://github.com/ballerinalang/ballerina/pull/3241/commits/1db472444007ae3a4d5f80b43be3711f1a71a3c2

Comment (lankavitharana) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/builder/BLangModelBuilder.java]: wrong commit id :)

Comment (maheeka) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/builder/BLangModelBuilder.java]: Fixed with https://github.com/ballerinalang/ballerina/pull/3241/commits/bd78dd8d4d2f7925fc42115d8af093e4c4838a67

Comment (maheeka) [inline modules/ballerina-core/src/main/java/org/ballerinalang/model/builder/BLangModelBuilder.java]: Fixed with https://github.com/ballerinalang/ballerina/pull/3241/commits/ad4abd9e72751503868314ebd03daf2fef6461db

**Pull request merged**